src/libraries/System.Threading.RateLimiting/src/System/Threading/RateLimiting/PartitionedRateLimiter.cs

```diff
@@ -29,268 +29,28 @@ public static class PartitionedRateLimiter
-                }
-            }
-
-            List<Exception>? aggregateExceptions = null;
-
-            // cachedLimiters is safe to use outside the lock because it is only updated by the Timer
-            foreach (KeyValuePair<TKey, Lazy<RateLimiter>> rateLimiter in _cachedLimiters)
-            {
-                if (!rateLimiter.Value.IsValueCreated)
-                {
-                    continue;
-                }
-                if (rateLimiter.Value.Value.IdleDuration is TimeSpan idleDuration && idleDuration > _idleTimeLimit)
-                {
-                    lock (Lock)
-                    {
-                        // Check time again under lock to make sure no one calls Acquire or WaitAsync after checking the time and removing the limiter
-                        idleDuration = rateLimiter.Value.Value.IdleDuration ?? TimeSpan.Zero;
-                        if (idleDuration > _idleTimeLimit)
-                        {
-                            // Remove limiter from the lookup table and mark cache as invalid
-                            // If a request for this partition comes in it will have to create a new limiter now
-                            // And the next time the timer runs the cache needs to be updated to no longer have a reference to this limiter
-                            _cacheInvalid = true;
-                            _limiters.Remove(rateLimiter.Key);
-
-                            // We don't want to dispose inside the lock so we need to defer it
-                            _limitersToDispose.Add(rateLimiter.Value.Value);
-                        }
-                    }
-                }
-                else if (rateLimiter.Value.Value is ReplenishingRateLimiter replenishingRateLimiter)
-                {
-                    try
-                    {
-                        replenishingRateLimiter.TryReplenish();
-                    }
-                    catch (Exception ex)
-                    {
-                        aggregateExceptions ??= new List<Exception>();
-                        aggregateExceptions.Add(ex);
-                    }
-                }
+                throw new ArgumentNullException(nameof(limiters));
             }
-
-            foreach (RateLimiter limiter in _limitersToDispose)
-            {
-                try
-                {
-                    await limiter.DisposeAsync().ConfigureAwait(false);
-                }
-                catch (Exception ex)
-                {
-                    aggregateExceptions ??= new List<Exception>();
-                    aggregateExceptions.Add(ex);
-                }
-            }
-            _limitersToDispose.Clear();
-
-            if (aggregateExceptions is not null)
+            if (limiters.Length == 0)
             {
-                throw new AggregateException(aggregateExceptions);
+                throw new ArgumentOutOfRangeException(nameof(limiters), "Must pass in at least 1 limiter.");
```

Comment:
Should just be `ArgumentException`? Does it really make sense for an array to be out of "range"?